ledger_device_sdk/src/nbgl.rs

```diff
@@ -0,0 +1,267 @@
+                fields.len()
+            } else {
+                CTX.review_pairs.len()
+            } as u8;
+
+            let confirm_string = "TRANSACTION\nSIGNED\0";
+            CTX.review_confirm_string[..confirm_string.len()]
+                .copy_from_slice(confirm_string.as_bytes());
+            let reject_string = "Transaction\nRejected\0";
+            CTX.review_reject_string[..reject_string.len()]
+                .copy_from_slice(reject_string.as_bytes());
+
+            let tag_value_list: nbgl_layoutTagValueList_t = nbgl_layoutTagValueList_t {
+                pairs: CTX.review_pairs.as_mut_ptr() as *mut nbgl_layoutTagValue_t,
+                callback: None,
+                nbPairs: CTX.nb_pairs,
+                startIndex: 2,
+                nbMaxLinesForValue: 0,
+                token: 0,
+                smallCaseForValue: false,
+                wrapping: false,
+            };
+
+            let sync_ret = ledger_secure_sdk_sys::sync_nbgl_useCaseTransactionReview(
+                &tag_value_list as *const nbgl_layoutTagValueList_t,
+                &icon as *const nbgl_icon_details_t,
+                "Review tx\0".as_ptr() as *const ::core::ffi::c_char,
+                "Subtitle\0".as_ptr() as *const ::core::ffi::c_char,
+                "Tx confirmed\0".as_ptr() as *const ::core::ffi::c_char,
+            );
+
+            match sync_ret {
+                ledger_secure_sdk_sys::NBGL_SYNC_RET_SUCCESS => {
+                    ledger_secure_sdk_sys::sync_nbgl_useCaseStatus(
+                        CTX.review_confirm_string.as_ptr() as *const ::core::ffi::c_char,
+                        true,
+                    );
+                    return true;
+                }
+                _ => {
+                    ledger_secure_sdk_sys::sync_nbgl_useCaseStatus(
+                        CTX.review_reject_string.as_ptr() as *const ::core::ffi::c_char,
+                        false,
+                    );
+                    return false;
+                }
+            }
+        }
+    }
+}
+
+// =================================================================================================
+
+impl NbglAddressConfirm {
+    pub fn verify_address(&mut self, address: &str) -> bool {
+        unsafe {
+            let icon = nbgl_icon_details_t {
+                width: 64,
+                height: 64,
+                bpp: 2,
+                isFile: true,
+                bitmap: CTX.icon.unwrap().as_ptr(),
+            };
+
+            let sync_ret = sync_nbgl_useCaseAddressReview(
+                address.as_ptr() as *const ::core::ffi::c_char,
+                &icon as *const nbgl_icon_details_t,
+                "Verify address\0".as_ptr() as *const ::core::ffi::c_char,
+                core::ptr::null(),
+            );
+
+            match sync_ret {
+                ledger_secure_sdk_sys::NBGL_SYNC_RET_SUCCESS => {
+                    return true;
```

Comment:
`sync_nbgl_useCaseStatus` should be added upon confirmation / rejection of address

Reply:
Done.